model/project_ref.go

```diff
@@ -2467,19 +2487,79 @@ func (p *ProjectRef) GetActivationTimeForVariant(variant *BuildVariant) (time.Ti
 		}
 	}
 
-	return defaultRes, nil
+	return versionCreateTime, nil
 }
 
-// GetActivationTimeForTask returns the time at which this task should next be activated.
-// Temporarily takes in the task ID that prompted this query, for logging.
-func (p *ProjectRef) GetActivationTimeForTask(t *BuildVariantTaskUnit) (time.Time, error) {
-	defaultRes := time.Now()
+// isActiveCronTimeRange checks that the proposed cron should activate now or
+// has already activated very recently.
+func (p *ProjectRef) isActiveCronTimeRange(proposedCron time.Time, now time.Time) bool {
+	const cronActiveRange = 5 * time.Minute
+	return !proposedCron.Before(now.Add(-cronActiveRange))
+}
+
+// isValidBVCron checks if a proposed time to activate a cron for a build
+// variant is valid. A cron scheduled to activate in the future is always valid,
+// but if the cron is scheduled to run in the past, that mean it will run
+// immediately. Crons scheduled for the past are only valid if they've recently
+// passed the proposed cron time and there's no conflicting cron that will
+// activate or has already activated.
+func (p *ProjectRef) isValidBVCron(bv *BuildVariant, proposedCron time.Time, now time.Time) (bool, error) {
```

Comment:
Since this is mostly the same function as `isValidTaskCron`, can we instead use one function for them or modularize [this part](https://github.com/evergreen-ci/evergreen/pull/8114/files#diff-dfa32e972797c36a4921f84900e1e4fefa32ea1a30c267cde0ec91ac72eaba7dR2507-R2523)?

Reply:
I tried consolidating into one helper function.